### PR TITLE
[Process] Dont test TTY if there is no TTY support

### LINKED ISSUE #38946: [Process] ProcessTest::testTTYCommand fails on macOS in non-interactive shells
**Symfony version(s) affected**: 4.4-dev

**Description**  
When running the Process test suite on macOS in a non-interactive shell, two tests fail with the following error message.
```
TTY mode requires /dev/tty to be read/writable.
```

This is for instance the case on GitLab's CI or when running the tests with PhpStorm.

**How to reproduce**  
Run the Process test suite in PhpStorm.

**Additional context**  
Discovered while working on #38620.

https://github.com/symfony/symfony/pull/38620/checks?check_run_id=1338904496#step:8:1025

```
There were 2 errors:

1) Symfony\Component\Process\Tests\ProcessTest::testTTYCommand
Symfony\Component\Process\Exception\RuntimeException: TTY mode requires /dev/tty to be read/writable.

/Users/runner/work/symfony/symfony/src/Symfony/Component/Process/Process.php:1075
/Users/runner/work/symfony/symfony/src/Symfony/Component/Process/Tests/ProcessTest.php:480

2) Symfony\Component\Process\Tests\ProcessTest::testTTYCommandExitCode
Symfony\Component\Process\Exception\RuntimeException: TTY mode requires /dev/tty to be read/writable.

/Users/runner/work/symfony/symfony/src/Symfony/Component/Process/Process.php:1075
/Users/runner/work/symfony/symfony/src/Symfony/Component/Process/Tests/ProcessTest.php:495
```

Bildschirmfoto 2020-11-01 um 18 40 55

### PR DESCRIPTION
**Pull request by Nyholm**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #38946
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Process/Tests/ProcessTest.php [hunk 1]
```diff
@@ -1438,7 +1446,7 @@ public function testRawCommandLine()
 (
     [0] => -
     [1] => a
-    [2] =>
```
Comment (derrabus) [inline src/Symfony/Component/Process/Tests/ProcessTest.php:1441]: Needs to be reverted. I've stepped into that trap multiple times myself.

**Comment by derrabus** [comment]
Well, this would make our tests pass. But my understanding is that the TTY detection that you use to skip the test is part of what we want to test here. Let's say we would break that detection logic. We would never know because the corresponding tests would flip from green to skipped. 😕

File: src/Symfony/Component/Process/Tests/ProcessTest.php (hunk 1 above)
Comment (Nyholm) [reply src/Symfony/Component/Process/Tests/ProcessTest.php:1441]: Oops, this is strange.. It is automatically "fixed" by .editorconfig. I did a permanent fix.

**Comment by Nyholm** [comment]
Is it testing `Process::isTtySupported()`? 

Isn't it testing that the `Process` is using `UnixPipes` and handle the exit code correctly?

File: src/Symfony/Component/Process/Tests/ProcessTest.php [hunk 2]
```diff
@@ -476,6 +476,10 @@ public function testTTYCommand()
             $this->markTestSkipped('Windows does not have /dev/tty support');
         }
 
+        if (!Process::isTtySupported()) {
+            $this->markTestSkipped('There is not TTY support');
```
Comment (fabpot) [inline src/Symfony/Component/Process/Tests/ProcessTest.php]: not -> no?

File: src/Symfony/Component/Process/Tests/ProcessTest.php [hunk 3]
```diff
@@ -491,6 +495,10 @@ public function testTTYCommandExitCode()
             $this->markTestSkipped('Windows does have /dev/tty support');
         }
 
+        if (!Process::isTtySupported()) {
+            $this->markTestSkipped('There is not TTY support');
```
Comment (fabpot) [inline src/Symfony/Component/Process/Tests/ProcessTest.php]: same here?

**Review by fabpot — CHANGES_REQUESTED** [review]

**Comment by Nyholm** [comment]
PR updated. Thank you

**Comment by derrabus** [comment]
@Nyholm If I change `Process::isTtySupported()` to always return `false`, those two tests are the only ones failing.

**Comment by Nyholm** [comment]
If `Process::isTtySupported()` always returns `false`, then these two tests will be skipped. 

If `Process::isTtySupported()` always returns true, yes, these will fail.

What alternative strategy do you have to solve #38946?

**Comment by derrabus** [comment]
> What alternative strategy do you have to solve #38946?

None. 😕

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @Nyholm.

**Pull request merged**